time/src/cli.rs

```diff
@@ -19,14 +65,70 @@ pub(crate) fn create_app<'a, 'b>() -> App<'a, 'b> {
                 .required(true),
         )
-        .arg(
-            Arg::with_name("posix")
-                .help(
-                    "Display time output in POSIX specified format as:\n\treal %f\n\tuser \
-                     %f\n\tsys  %f\nTimer accuracy is arbitrary, but will always be counted in \
-                     seconds.",
-                )
-                .long("posix")
-                .short("p"),
-        )
+        .arg(Arg::with_name("posix").help(POSIX_FMT_HELP).long("posix").short("p"));
+    configure_extensions(app)
+}
+
+#[rustfmt::skip]
+fn configure_extensions<'a, 'b>(app: App<'a, 'b>) -> App<'a, 'b> {
+    let use_csh_fmt = Arg::with_name("use_csh_fmt")
+        .conflicts_with_all(&["posix"])
+        .help(CSH_FMT_HELP)
+        .long("csh-format")
+        .short("c");
+    let format_string = Arg::with_name("format_string")
+        .conflicts_with_all(&["posix", "use_csh_fmt"])
+        .takes_value(true)
+        .validator(|s: String| -> Result<(), String> {
+            if s.is_empty() || !s.is_ascii() {
+                Err(String::from("Format string must be non-empty ASCII"))
+            } else {
+                Ok(())
+            }
+        })
+        .help(FMT_ARG_HELP)
+        .long("format")
+        .short("f");
+    let use_tcsh_fmt = Arg::with_name("use_tcsh_fmt")
+        .conflicts_with_all(&["posix", "use_csh_fmt", "format_string"])
+        .help(TCSH_FMT_HELP)
+        .long("tcsh-format")
+        .short("t");
+
+    let dump_rusage = Arg::with_name("dump_rusage")
+        .conflicts_with_all(&["posix"])
+        .help("Lists resource utilization information. The contents of \
+              the\ncommand process's rusage structure are printed")
+        .long("rusage")
+        .short("l");
+
+    let human_readable = Arg::with_name("human_readable")
+        .conflicts_with("posix")
+        .help("Time durations are printed in hours, minutes, seconds")
+        .long("human-readable")
+        .short("h");
+
+    let output_path = Arg::with_name("output_file")
+        .takes_value(true)
+        .help("Write the output to file instead of stderr.\
+              If file exists and the -a flag is not specified,\
+              the file will be overwritten.")
+        .long("output-path")
+        .short("o");
+
+    let append_output = Arg::with_name("append_mode")
+        .help("If the -o flag is used, append to specified file rather that\
+              overwrite it. Otherwise this option has no effect")
+        .long("append-mode")
+        .short("a");
+
+    if cfg!(target_os = "netbsd") {
+        app.args(&[use_csh_fmt, format_string, dump_rusage, use_tcsh_fmt])
+    } else if cfg!(any(target_os = "freebsd", target_os = "dragonflybsd")) {
+        app.args(&[append_output, dump_rusage, human_readable, output_path])
+    } else if cfg!(any(target_os = "openbsd", target_os = "macos")) {
+        app.arg(dump_rusage)
+    } else {
+        app
+    }
```

Comment:
There's nothing platform specific about the individual flags (except for `-l`, which this PR doesn't implement anyway). They're only gated right now because the original binaries have varying feature-set across different platforms. I think it would be better to merge them anyway since it makes it easier to test the binary